### DISCUSSION TITLE
Get Inputparameters from anthor material

### DISCUSSION THREAD

**Discussion by LifengGan** [idaholab/moose#25810] [Q&A Modules: Solid mechanics] [answered]
Dear moose developers,

I am developing code based on `ComputeCrackedStress`, and not quite well understand the logic on how to pass the material parameters from other material.

It use `getMaterialPropertyByName `to fetch the stress, and strain from stress model.

But if i want to get material parameters from other model, for examples,` _number_slip_systems` from a crystal plasticity mode

Is it the same way by using the `getMaterialPropertyByName`?

Thanks.

**Comment by GiudGiud** [comment]
Hello

Unless `_number_slip_systems` is declared as material property itself, then no it wont work with getMaterialPropertyByName.
One of the cannon way to do this is to make that attribute a material property as well.

We dont have a way to 'forward' input parameters from one object to another. 
if you obtain a pointer to the first material using this routine on the FEProblem
```
  std::shared_ptr<MaterialBase> getMaterial(std::string name,
                                            Moose::MaterialDataType type,
                                            THREAD_ID tid = 0,
                                            bool no_warn = false);
```
AND
write a getter on that material for the attribute you want, then you can dynamic_past that first pointer to the type needed and call the getter. 

Guillaume

**Reply by GiudGiud** [reply]
for that particular attribute, it might be stored elsewhere than the material as well. I reclassified the post under tensor mechanics to see if others know

**Reply by sapitts** [reply, marked as answer]
The `_number_slip_systems` is stored only as an unsigned integer, not as a material property. You could, however, query the length of one of the `MaterialProperty` vectors that contains a value for each slip system to get the same information.  As an example the vector `_tau[_qp]` can be retrieved through `getMaterialPropertyOld<std::vector<Real>>(_base_name + "applied_shear_stress")`

Hope this helps,
Stephanie
